Fix problems with epub embedded fonts (BL-4146)

andrew-polk opened the pull request:
We weren't including fonts for styles
We were embedding extra fonts

This was cherry-picked from 3.8 but then changed to implement the includeFallbackFonts flag.

(cherry picked from commit d6ae7e9c6f993187cc2fcdbff3dd392411738702)

Conflicts:
-	src/BloomExe/Book/HtmlDom.cs
-	src/BloomExe/CLI/GetUsedFontsCommand.cs
-	src/BloomExe/Publish/EpubMaker.cs
-	src/BloomTests/Publish/ExportEpubTests.cs

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloomdesktop/1428)
<!-- Reviewable:end -->

JohnThomson commented:
The comment implies we're doing something different about fall-back fonts in 3.7, but I don't see what. Seems just like 3.8: only include the first font in each sequence.
Other than that I haven't looked very closely...we would already have reviewed this change in 3.8.

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1428#-:-KbR2emP3t6t1YroAlqq:b-955s13)*
<!-- Sent from Reviewable.io -->

andrew-polk commented:
The intention is to do in 3.7 exactly as we do in 3.8. The difference was that the includeFallbackFonts flag already existed in 3.8, so we changed it, but it didn't exist in 3.7, so I implemented it.
I just wanted to get some eyes on it before it went into 3.7 since it was not as straightforward as cherrypick the commit.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1428#-:-KbR5xtSVNaYVNn0oqct:bm56kiz)*
<!-- Sent from Reviewable.io -->

JohnThomson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1428#-:-KbVbQXpoB0oadaKc4G6:bnfp4nl)*
<!-- Sent from Reviewable.io -->
